File: LC Mothly/MARCH/insertInterval.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class insertInterval {
    public static void main(String[] args) {
        int[][] intervals = { { 1, 3 }, { 6, 9 } };
        int[] newInterval = { 2, 5 };
        int[][] result = insert(intervals, newInterval);
        // for (int i = 0; i < result.length; i++) {
        //     System.out.println(result[i][0] + " " + result[i][1]);
        // }
System.out.println(Arrays.deepToString(result));
    }

    public static int[][] insert(int[][] intervals, int[] newInterval) {
        List<int[]> result = new ArrayList<>();
        int i = 0;
        while (i < intervals.length && intervals[i][1] < newInterval[0]) {
            result.add(intervals[i]);
            i++;
        }
        while (i < intervals.length && intervals[i][0] <= newInterval[1]) {
            newInterval[0] = Math.min(newInterval[0], intervals[i][0]);
            newInterval[1] = Math.max(newInterval[1], intervals[i][1]);
            i++;
        }
        result.add(newInterval);
        while (i < intervals.length) {
            result.add(intervals[i]);
            i++;
        }
        return result.toArray(new int[result.size()][]);
        
    }
}

File: Topic Wise/Intervals/insertInterval.py
intervals = [[1,2],[3,5],[6,7],[8,10],[12,16]]
newInterval = [4,8]

def insert(intervals,newInterval):
    intervals.append(newInterval)
    intervals.sort(key=lambda x:x[0])

    merged=[intervals[0]]

    for i in range(1,len(intervals)):
        if merged[-1][1]>=intervals[i][0]:
            merged[-1][1]=max(merged[-1][1],intervals[i][1])
        else:
            merged.append(intervals[i])

    return merged

print(insert(intervals,newInterval))